### PR TITLE
Add `workload_identity_labels` to RoleCondition

### PR DESCRIPTION
**Pull request by strideynet**
Part of https://github.com/gravitational/teleport/issues/49574
As per RFD 191: https://github.com/gravitational/teleport/pull/49133

Adds the `workload_identity_labels` label matcher to role conditions. This will be used to grant/restrict access to issuing workload credentials using the WorkloadIdentity resource.

This is a pre-req of a later PR that will introduce the IssueWorkloadIdentity RPC.

### CODE REVIEW

**Review by hugoShaka — APPROVED** [review]

**Review by ptgott — APPROVED** [review]

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -3439,6 +3439,15 @@ message RoleConditions {
     (gogoproto.nullable) = false,
     (gogoproto.jsontag) = "github_permissions,omitempty"
   ];
+
+  // WorkloadIdentityLabels controls whether or not specific WorkloadIdentity
+  // resources can be invoked. Further authorization controls exist on the
+  // WorkloadIdentity resource itself.
+  wrappers.LabelValues WorkloadIdentityLabels = 44 [
```
Comment (tigrato) [inline api/proto/teleport/legacy/types/types.proto:3446]: Should we also add role expressions? All resources have them

**Review by tigrato — COMMENTED** [review]
requesting a change here

File: api/proto/teleport/legacy/types/types.proto (hunk 1 above)
Comment (hugoShaka) [reply api/proto/teleport/legacy/types/types.proto:3446]: +1 for consistency, it's terribly frustrating when we support expressions on every field but the one I try

Comment (strideynet) [reply api/proto/teleport/legacy/types/types.proto:3446]: 👍  Sure - easy enough to do. I wasn't sure how widespread their use was.

Comment (strideynet) [reply api/proto/teleport/legacy/types/types.proto:3446]: [65ed124](https://github.com/gravitational/teleport/pull/50071/commits/65ed124f872f33767db2a2e1e6ba29c9e7fff478)

**Pull request merged**